informer/informer.go

```diff
@@ -224,18 +229,29 @@ func (i *Informer) Watch(ctx context.Context, endpoint string, factory ZeroObjec
 	return deleteChan, updateChan, errChan
 }
 
-// cacheAndSend sends the provided event object to the provided update channel
-// and caches the provided event object.
-func (i *Informer) cacheAndSend(event watch.Event, updateChan chan watch.Event) error {
-	updateChan <- event
-
+// cacheAndSend stores the provided event object in the informer cache and
+// dispatches it based on its properties. cacheAndSend sends the provided event
+// object to the provided update channel in case the event object has no
+// deletion timestamp. In case the deletion timestamp of the provided event
+// object is non-nil, it is send to the provided delete channel.
+func (i *Informer) cacheAndSend(event watch.Event, deleteChan, updateChan chan watch.Event) error {
 	k, err := cache.MetaNamespaceKeyFunc(event.Object)
 	if err != nil {
 		return microerror.Mask(err)
 	}
-
 	i.cache.Store(k, event)
 
+	m, err := meta.Accessor(event.Object)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	t := m.GetDeletionTimestamp()
+	if t == nil {
+		updateChan <- event
+	} else {
+		deleteChan <- event
+	}
```

Comment:
This is a result of testing finalizers with CRDs on 1.8. When a event object is deleted and it has a finalizer, it is replayed on resync period and then send to the delete channel.